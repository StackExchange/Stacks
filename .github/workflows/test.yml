name: Build and Test

on:
  workflow_call:
    inputs:
      command:
        description: 'npm scripts command to execute'
        type: string
        required: true
      command_description:
        description: 'Description of the npm scripts command to execute'
        type: string
        required: true
      needs_playwright:
        description: 'Whether or not to install playwright'
        type: boolean
        required: true

jobs:
  build-and-test:
    name: ${{ inputs.command_description }}
    runs-on: ubuntu-latest
    steps:
      - name: ⬇️ Checkout
        uses: actions/checkout@v3
        with:
          lfs: true

      - name: ⎔ Setup node
        uses: actions/setup-node@v3
        with:
          node-version: 'lts/*'

      - name: 🏗 Install Dependencies
        run: npm ci

      - name: 🏗 Install Playwright
        if: inputs.needs_playwright == 'true'
        run: npx playwright install --with-deps

      - name: ▶️ ${{ inputs.command_description }}
        run: ${{ inputs.command }}

      - name: ⬆️ Upload Visual Regression Test Results
        uses: actions/upload-artifact@v3
        if: ${{ failure() && inputs.command_description == 'Visual Regression Tests' }}
        with:
          name: visual-regression-test-results
          path: screenshots

concurrency:
  group: ${{ github.head_ref || github.run_id }}
  cancel-in-progress: true
