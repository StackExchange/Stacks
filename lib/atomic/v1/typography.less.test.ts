import { describe, it, expect } from "vitest";
import { renderLess } from "../../test/less-test-utils";

describe("atomic: border v1", () => {
    it("should output all v1 atomic css classes for colors", async () => {
        const css = await renderLess(`
            @import (reference) "./lib/exports/v1/constants-colors.less";
            @import (reference) "./lib/exports/mixins.less";
            @import "./lib/atomic/v1/typography.less";
        `);

        expect(css).toMatchSnapshot();
    });
});
