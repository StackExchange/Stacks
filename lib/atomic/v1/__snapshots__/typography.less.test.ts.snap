// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`atomic: border v1 > should output all v1 atomic css classes for colors 1`] = `
".fc-dark-legacy {
    color: var(--fc-dark-legacy) !important;
}

.fc-medium-legacy {
    color: var(--fc-medium-legacy) !important;
}

.fc-light-legacy {
    color: var(--fc-light-legacy) !important;
}
"
`;
